File: feb/4 feb/lc63.py
from typing import *

class Solution:
    def uniquePathsWithObstacles(self, obstacleGrid: List[List[int]]) -> int:
        n,m = len(obstacleGrid),len(obstacleGrid[0])
        mat = [[0]*m for i in range(n)]
        for i in range(n):
            for j in range(m):
                if obstacleGrid[i][j] == 1:
                    mat[i][j] = 0
                elif i == 0 and j == 0:
                    mat[i][j] = 1
                elif i == 0:
                    mat[i][j] = mat[i][j-1]
                elif j == 0:
                    mat[i][j] = mat[i-1][j]
                else:
                    mat[i][j] = mat[i][j-1] + mat[i-1][j]
        return mat[n-1][m-1]

File: june/1june/lc63.java
class Solution {
    public int uniquePathsWithObstacles(int[][] grid) {
        int n = grid.length,m=grid[0].length;
        int[][] path = new int[n][m];
        for(int i=0;n>i;i++){
            for(int j=0;m>j;j++){
                int c = 0;
                if(grid[i][j] == 1) continue;
                if(i == 0 && j == 0) c = 1;
                if(i != 0) c = path[i-1][j];
                if(j != 0) c += path[i][j-1];
                path[i][j] = c;
            }
        }
        return path[n-1][m-1];
    }
}